osu.Game/Online/RealtimeMultiplayer/MultiplayerRoomSettings.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using osu.Game.Online.API;
+
+namespace osu.Game.Online.RealtimeMultiplayer
+{
+    [Serializable]
+    public class MultiplayerRoomSettings : IEquatable<MultiplayerRoomSettings>
+    {
+        public int? BeatmapID { get; set; }
+
+        public int? RulesetID { get; set; }
+
+        [NotNull]
+        public IEnumerable<APIMod> Mods { get; set; } = Enumerable.Empty<APIMod>();
+
+        public bool Equals(MultiplayerRoomSettings other) => BeatmapID == other.BeatmapID && Mods.SequenceEqual(other.Mods) && RulesetID == other.RulesetID;
+
+        public override string ToString() => $"Beatmap:{BeatmapID} Mods:{string.Join(',', Mods)} Ruleset:{RulesetID}";
+
+        public static MultiplayerRoomSettings Empty() => new MultiplayerRoomSettings();
```

Comment:
A bit curious why this is a thing. I'd probably either remove this or hide the parameterless ctor by making it `private`, so that there's no ambiguity whether the ctor and this static factory method differ in any capacity.

Reply:
parameterless needs to be there for serialization i belied. will just remove this since it doesn't add much clarity.